src/App.jsx

```diff
@@ -3,46 +3,62 @@ import React, { useState, useEffect } from 'react';
 import AddTodoForm from './AddTodoForm';
 import TodoList from './TodoList';
 
-function useSemiPersistentState() {
-  const [todoList, setTodoList] = useState(
-    JSON.parse(localStorage.getItem('savedTodoList')) || []
-  );
-
-  useEffect(() => {
-    localStorage.setItem('savedTodoList', JSON.stringify(todoList));
-  }, [todoList]);
-
-  return [todoList, setTodoList];
-}
-
 function App() {
-  const [todoList, setTodoList] = useSemiPersistentState();
+  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem('todoList')) ?? []);
+  const [isLoading, setIsLoading] = useState(true);
 
+  // Load initial data and save to localStorage
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const url = `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`;
+        const options = {
+          method: 'GET',
+          headers: {
+            Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`
+          }
+        };
+        const response = await fetch(url, options);
+        if (!response.ok) {
+          throw new Error('Error: ${response.status}');
+        }
+        const data = await response.json();
+        const todos = data.records.map(record => ({
+          id: record.id,
+          title: record.fields.title,
+          createTime: record.createdTime
+        }));
+        //console.log('Todos:', todos); 
+        setTodoList(todos);
+        localStorage.setItem('todoList', JSON.stringify(todos)); //save data, localstorage
```

Comment:
This approach works when you first render your app. But what if you add a new item to your application? Have you tested adding new items and then checked to see if they show up in `localStorage`?

Reply:
Change the approach, every time I add or delete a task, the todoList state will be updated and the updated data will be saved to local storage. `  // Funtion to add tasks
  const addTodo = (newTodo) => {
    setTodoList(prevTodoList => {
      const updatedTodoList = [...prevTodoList, newTodo];
      localStorage.setItem('todoList', JSON.stringify(updatedTodoList));
      return updatedTodoList;
    });
  };

  // Funtion to remove a todo
  const removeTodo = (id) => {
    const updatedTodoList = todoList.filter(todo => todo.id !== id);
    setTodoList(updatedTodoList);
    localStorage.setItem('todoList', JSON.stringify(updatedTodoList));
  };`